src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -775,37 +790,124 @@ static BOOL AttributeHasChangesOmittedQuality(MTRAttributePath * attributePath)
     // 4. Cache has no entry
     // TODO: add option for BaseSubscriptionCallback to report during priming, to reduce when case 4 is hit
     if (!attributeIsSpecified || ![self _subscriptionAbleToReport] || hasChangesOmittedQuality || !attributeValueToReturn) {
+        // Read requests container will be a mutable array of items, each being an array containing:
+        //   [endpoint ID, cluster ID, attribute ID, params]
+        // Batching handler should only coalesce when params are equal.
+
+        // For this single read API there's only 1 array item. Use NSNull to stand in for nil params for easy comparison.
+        NSArray * readRequestData = @[ endpointID, clusterID, attributeID, params ?: [NSNull null] ];
```

Comment:
Thinking about future bits a bit, why not just store a pair: MTRAttributeRequestPath and params?  That will extend to wildcards much more easily if we add those in the future.

Reply:
I've filed https://github.com/project-chip/connectedhomeip/issues/27025 , and we can do the refactor in there.

Reply:
Sure why not 👍 Done.